dll/win32/shell32/precomp.h

```diff
@@ -166,12 +194,28 @@ static inline UINT SeeFlagsToCmicFlags(UINT flags)
 
 // CStubWindow32 --- The owner window of file property sheets.
 // This window hides taskbar button of property sheet.
+#define CSTUBWINDOW32_CLASSNAME _T("StubWindow32")
 class CStubWindow32 : public CWindowImpl<CStubWindow32>
 {
 public:
-    DECLARE_WND_CLASS_EX(_T("StubWindow32"), 0, COLOR_WINDOWTEXT)
+    DECLARE_WND_CLASS_EX(CSTUBWINDOW32_CLASSNAME, 0, COLOR_WINDOWTEXT)
+    enum {
+        TYPE_FORMATDRIVE = 1,
+        TYPE_PROPERTYSHEET,
```

Comment:
The format dialog is identified (used to switch to an existing instance if there is one) by TYPE_FORMATDRIVE + "C:" (or whatever the drive letter is) and the property sheet is TYPE_PROPERTYSHEET + the parsing path of the PIDL. However, these two things are not stored together, the "path" is stored in the window caption of the stub window to avoid having to deal with the memory for it. And the type fits in the `SetProp` handle value.

If you want the callers to provide a full unique string they would have to create a new extra string for no reason and what would be the format? `wsprintf(buffer, L"%s:%s", L"PropertySheet", path)`?

Or we could use a unique `SetProp` string for each type but that saves us nothing (we would just do `SetProp(hWnd, name, 1)` for each then instead of a specific type value).